pylint/checkers/base/basic_checker.py

```diff
@@ -657,13 +657,38 @@ def _check_misplaced_format_function(self, call_node: nodes.Call) -> None:
             ):
                 self.add_message("misplaced-format-function", node=call_node)
 
+    @staticmethod
+    def _is_terminating_func(node: nodes.Call) -> bool:
+        """Detect call to exit(), quit(), os._exit(), or sys.exit()."""
+        if (
+            not isinstance(node.func, nodes.Attribute)
+            and not (isinstance(node.func, nodes.Name))
+            or isinstance(node.parent, nodes.Lambda)
+        ):
+            return False
+
+        qnames = {"_sitebuiltins.Quitter", "sys.exit", "posix._exit", "nt._exit"}
+
+        try:
+            for inferred in node.func.infer():
+                if hasattr(inferred, "qname") and inferred.qname() in qnames:
+                    return True
```

Comment:
I'll be the first to say using hasattr isn't ideal and I did attempt isinstance, however:

for the cases of node being built builtin exit or quit, these get an inferred instance if :

```
(Pdb) inferred
<Instance of _sitebuiltins.Quitter at 0x4428041360>
(Pdb) from _sitebuiltins import Quitter
(Pdb)  isinstance(inferred, Quitter)
False
```

so as much as I tried to do `if isinstance(inferred, (nodes.FunctionDef, Quitter)` or something similar, it never worked. so hasattr was the best I could get to work

Reply:
This seems to be a case of `astroid.Instance` right? We could just instance check for that?

If possible I'd like us to do a follow-up, since the `hasattr` doesn't provide actual type safety..